axum/src/extract/query.rs

```diff
@@ -62,6 +62,37 @@ where
     }
 }
 
+impl<T> Query<T>
+where
+    T: DeserializeOwned,
+{
+    /// Attempts to construct a [`Query`] from a reference to a [`Uri`].
+    ///
+    /// # Example
+    /// ```
+    /// use axum::extract::Query;
+    /// use http::Uri;
+    /// use serde::Deserialize;
+    ///
+    /// #[derive(Deserialize)]
+    /// struct ExampleParams {
+    ///     foo: String,
+    ///     bar: u32,
+    /// }
+    ///
+    /// let uri: Uri = "http://example.com/path?foo=hello&bar=42".parse().unwrap();
+    /// let result: Query<ExampleParams> = Query::from_uri(&uri).unwrap();
+    /// assert_eq!(result.foo, String::from("hello"));
+    /// assert_eq!(result.bar, 42);
+    /// ```
+    pub fn from_uri(value: &Uri) -> Result<Self, QueryRejection> {
+        let query = value.query().unwrap_or_default();
+        let params: T =
+            serde_urlencoded::from_str(query).map_err(FailedToDeserializeQueryString::from_err)?;
+        Ok(Query(params))
+    }
```

Comment:
I think we should still include the `try_` prefix in the name.

```suggestion
    pub fn try_from_uri(value: &Uri) -> Result<Self, QueryRejection> {
        let query = value.query().unwrap_or_default();
        let params =
            serde_urlencoded::from_str(query).map_err(FailedToDeserializeQueryString::from_err)?;
        Ok(Query(params))
    }
```

We should also call this from `Query::from_request`.

Reply:
Just to check if I understand, do you also want me to implement the `FromRequest` trait for `Query`?

Reply:
Sorry no, `FromRequestParts`. The implementation is currently duplicated. It shouldn't be.